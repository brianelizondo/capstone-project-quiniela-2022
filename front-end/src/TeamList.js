import React, { useState, useEffect } from 'react';

// import APIFootball api
import APIFootball from "./api-football";

import TeamCard from './TeamCard';

function TeamList(){
    // initial state for quinielas list
    const [teams, setTeams] = useState([]);

    useEffect(() => {
        async function getTeams() {
            const resp = await APIFootball.getTeams();
            setTeams(resp);
        }
        getTeams();
    }, []);

    return (
        <div className="TeamList col-md-8 offset-md-2">
            <h1>Teams in FIFA World Cup 2022</h1>
            <p>List of all teams competing</p>
            
            <div className="TeamList-list">
                { teams.map(team => 
                <div key={team.shortName} className="TeamList-list-card"><TeamCard key={team.shortName} team={team} /></div>
                )}
            </div>
        </div>
    );
}

export default TeamList;