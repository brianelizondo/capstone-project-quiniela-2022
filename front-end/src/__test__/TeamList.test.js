import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import '@testing-library/jest-dom';
import { MemoryRouter } from "react-router-dom";

import TeamList from '../TeamList';

// Smoke Test
describe("Smoke Test", () => {
    it("renders 'TeamList' component without crashing", () => {
        render(
            <MemoryRouter initialEntries={["/teams"]}>
                <TeamList />
            </MemoryRouter>
        );
        const textFound = screen.getByText("Teams in FIFA World Cup 2022");
        expect(textFound).toBeInTheDocument();
    });
});

// Snapshot Tests
describe("Snapshot Test", () => {
    it("'TeamList' matches snapshot", () => {
        const {asFragment} = render(
            <MemoryRouter initialEntries={["/teams"]}>
                <TeamList />
            </MemoryRouter>
        );
        expect(asFragment()).toMatchSnapshot();
    });
});

describe("Additional Test", () => {
    it("works when you click on a team card go to the team details", async () => {
        render(
            <MemoryRouter initialEntries={["/teams"]}>
                <TeamList />
            </MemoryRouter>
        );
        const user = userEvent.setup()

        waitFor(() => { expect(screen.getByText("List of all teams competing")).toBeInTheDocument() });
        waitFor(async () => { await user.click(screen.getAllByTestId("team-")) });
        waitFor(() => { expect(screen.getByText("Details for the team")).toBeInTheDocument() });
    });
});