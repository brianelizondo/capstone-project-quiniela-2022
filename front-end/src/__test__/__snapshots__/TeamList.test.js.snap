// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Test 'TeamList' matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="TeamList col-md-8 offset-md-2"
  >
    <h1>
      Teams in FIFA World Cup 2022
    </h1>
    <p>
      List of all teams competing
    </p>
    <div
      class="TeamList-list"
    />
  </div>
</DocumentFragment>
`;
